src/main/java/duke/CommandHandler.java

```diff
@@ -0,0 +1,276 @@
+                    System.out.println(":( OOPS!!! This task does not exist.");
+                    drawLine();
+                }
+            }
+            else {
+                createNewTask(command);
+            }
+            try {
+                df.saveFile(df.getFile(), getTaskListAsString());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        } while (command != "bye");
+    }
+
+    private static String getTaskListAsString() {
+        String taskList = "";
+        for (int i = 0; i < taskCounter; ++i) {
+            char taskType = tasks.get(i).getType();
+            taskList += tasks.get(i).getType() + " | ";
+            taskList += (tasks.get(i).isDone() ? "1" : "0") + " | ";
+            switch(taskType) {
+            case 'T':
+                taskList += tasks.get(i).getName() + "\n";
+                break;
+            case 'D':
+            case 'E':
+                taskList += tasks.get(i).getName() + " | " + tasks.get(i).getDateTime() + "\n";
+                break;
+            default:
+                System.out.println("Task list in invalid format");
+            }
+        }
+        return taskList;
+    }
+    private static void loadTasks(DukeFile df) {
+        try {
+            String taskList = df.loadFile(df.getFile());
+            while(taskList.length() > 0 && taskList.charAt(taskList.length()-1) == '/') {
+                String taskName, taskDateTime;
+                int ind_pipe;
+                int ind_backslash = taskList.indexOf("/");
+                String task = taskList.substring(0, ind_backslash);
+                taskList = taskList.substring(ind_backslash + 1);
+                char taskType = task.charAt(0);
+                boolean isDone = task.charAt(4) == '1';
+                switch(taskType) {
+                case 'T':
+                    taskName = task.substring(8);
+                    tasks.add(taskCounter, new ToDo(taskName, isDone, taskType));
+                    taskCounter++;
+                    break;
+                case 'D':
+                    task = task.substring(8);
+                    ind_pipe = task.indexOf("|");
+                    taskName = task.substring(0, ind_pipe - 1);
+                    taskDateTime = task.substring(ind_pipe + 2);
+                    tasks.add(taskCounter, new Deadline(taskName, isDone, taskType, taskDateTime));
+                    taskCounter++;
+                    break;
+                case 'E':
+                    task = task.substring(8);
+                    ind_pipe = task.indexOf("|");
+                    taskName = task.substring(0, ind_pipe - 1);
+                    taskDateTime = task.substring(ind_pipe + 2);
+                    tasks.add(taskCounter, new Event(taskName, isDone, taskType, taskDateTime));
+                    taskCounter++;
+                    break;
```

Comment:
[Code Quality] Each of the case logic could probably be extracted out to improve SLAP.